src/controller/tests/TestAutoCommissioner.cpp

```diff
@@ -338,4 +338,113 @@ TEST_F(AutoCommissionerTest, NextStageConfigureTCAcknowledgments)
     EXPECT_EQ(nextStage, kSendPAICertificateRequest);
 }
 
+TEST_F(AutoCommissionerTest, TrySecondaryNetwork_confirm_true)
+{
+    AutoCommissionerTestAccess privateConfigCommissioner(&mCommissioner);
+    privateConfigCommissioner.TrySecondaryNetwork();
+    EXPECT_EQ(privateConfigCommissioner.TryingSecondaryNetwork(), true);
+}
+
+TEST_F(AutoCommissionerTest, ResetTryingSecondaryNetwork_confirm_false)
+{
+    AutoCommissionerTestAccess privateConfigCommissioner(&mCommissioner);
+    privateConfigCommissioner.ResetTryingSecondaryNetwork();
+    EXPECT_EQ(privateConfigCommissioner.TryingSecondaryNetwork(), false);
+}
+TEST_F(AutoCommissionerTest, IsScanNeededCombinations)
+{
+    struct Case
+    {
+        const char * name;
+        bool attemptWiFi;
+        bool attemptThread;
+        EndpointId wifiEndpoint;
+        EndpointId threadEndpoint;
+        bool expected;
+    };
+
+    Case cases[] = {
+        { "WiFiAndThreadSet", true, true, 0x000, 0x000, true },
```

Comment:
bools and numbers in sequence are hard to follow. Could you use the `.` initializer here to make this faster to read? like:

```cpp
{
   .name = "WifiAndThreadSet",
   .attemptWiFi = true,
   .attemptThread = true,
   .wifiEndpoint = 0,
   .threadEndpoint = 0,
   .expected = true,
}
```